tests/network/primary_pod_network.go

```diff
@@ -70,25 +70,69 @@ var _ = SIGDescribe("[Serial]Primary Pod Network", func() {
 		})
 
 		Context("VMI connected to the pod network using bridge binding", func() {
-			var vmi *v1.VirtualMachineInstance
+			When("Guest Agent exists", func() {
+				var (
+					vmi   *v1.VirtualMachineInstance
+					vmiIP = func() (string, error) {
+						vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+						if err != nil {
+							return "", err
+						}
+						return vmi.Status.Interfaces[0].IP, nil
+					}
 
-			BeforeEach(func() {
-				vmi = setupVMI(virtClient, vmiWithBridgeBinding())
-			})
+					vmiIPs = func() ([]string, error) {
+						vmi, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+						if err != nil {
```

Comment:
please Assert in those two funcs if vm isnt found, else you might mask a problem

Reply:
If vm is not found `Get` will fail and then we return the error, that's more than enough.

Reply:
it will still run `Eventually` since its called eventually
masking potential problems and consuming time
this is what we are doing everywhere, using assert in Eventually on the ones we dont want to mask

Reply:
What oshoval is pointing out is that at this point (also for `vmiIP` func), a VMI should exist for sure and if it does not, there is no point in retrying (as something really bad happened). We should retry in order to get a refreshed IP information, anything else is an error and we should exit immediately. The only way to exit immediately in an `Eventually` loop is to assert.

Bottom line, seems right to assert here on the error.
Is there any reason you think we should keep retrying?

Reply:
ok, got it now, will assert if it does not exist.

Reply:
Done

Reply:
Why do you consider a failure (any failure) to fetch the VMI object to merit a retry? The retry here is to get an updated VMI status, not to mask errors on the API (or due to other causes).

Could you please explain?

Reply:
Right, let me just assert any error.